src/Components/test/E2ETest/ServerExecutionTests/ServerComponentRenderingTest.cs

```diff
@@ -36,5 +35,32 @@ public void ThrowsIfRenderIsRequestedOutsideSyncContext()
                 $"{typeof(InvalidOperationException).FullName}: The current thread is not associated with the renderer's synchronization context",
                 () => result.Text);
         }
+
+        [Fact]
+        public void ReconnectUI()
+        {
+            MountTestComponent<DispatchingComponent>();
+            var selector = By.CssSelector("div.modal");
+
+            var element = Browser.FindElement(selector);
+            Assert.False(element.Displayed);
+
+            var javascript = (IJavaScriptExecutor)Browser;
+            javascript.ExecuteScript(@"
+window.modalDisplayState = [];
+window['Blazor'].circuitHandlers.push({
```

Comment:
Super-minor nit: the `window['Blazor']` syntax is only to make the TypeScript compiler happy. In JavaScript code, this can be written more simply as `window.Blazor` or even just `Blazor`.